service/history/execution/context.go

```diff
@@ -265,11 +267,18 @@ func NewContext(
 }
 
 func (c *contextImpl) Lock(ctx context.Context) error {
+	c.lockTime = time.Now()
 	return c.mutex.Lock(ctx)
 }
 
 func (c *contextImpl) Unlock() {
+	elapsed := time.Since(c.lockTime)
 	c.mutex.Unlock()
+	c.metricsClient.RecordTimer(metrics.WorkflowContextScope, metrics.WorkflowContextLockLatency, elapsed)
+	if elapsed > c.maxLockDuration {
+		c.maxLockDuration = elapsed
+		c.logger.Info("workflow context lock is released. this is logged only when it's longer than maxLockDuration", tag.WorkflowContextLockLatency(elapsed))
```

Comment:
I think this could end up being somewhat spammy, since it'll repeatedly log on a per-context basis until it eventually reaches a maximum that isn't exceeded. What do you think of initializing it to a moderately high constant (1-5s maybe)?

Also since we update this field (and elapsed) outside of holding the lock I don't think this value will always be accurate. A second call to Lock() is going to reset `lockTime`.

Reply:
I have no idea where that not-noisy value is (1-5s) so decided to go this way. Let me initialize it to 1s so quick scenarios do not cause startup log noise